client/src/components/FormInputField.tsx

```diff
@@ -0,0 +1,36 @@
+import { InputHTMLAttributes } from "react";
+import { FieldError, UseFormRegisterReturn } from "react-hook-form";
+
+interface FormInputFieldProps {
+  register: UseFormRegisterReturn;
+  label?: string;
+  error?: FieldError;
+  inputGroupElement?: JSX.Element;
+}
+
+export default function FormInputField({
+  register,
+  label,
+  error,
+  inputGroupElement,
+  ...props
+}: FormInputFieldProps & InputHTMLAttributes<HTMLInputElement>) {
+  return (
+    <div className="mb-4">
+      {label && <label className="block text-gray-700 mb-2">{label}</label>}
+      <div className="flex items-center">
+        <input
+          {...register}
+          {...props}
+          className={`block w-full max-w-md bg-transparent outline-none border-b-2 py-2 px-4 ${
+            error
+              ? "text-red-500 border-red-600"
+              : "text-gray-900 border-purple-900"
+          }`}
+        />
+        {inputGroupElement}
+      </div>
+      {error && <p className="text-sm text-red-500 mt-1">{error.message}</p>}
```

Comment:
make it more robust, you might want to ensure that error.message exists before rendering:

`{error?.message && <p className="text-sm text-red-500 mt-1">{error.message}</p>}`